lib/library.rb

```diff
@@ -0,0 +1,49 @@
+class Library
+  attr_reader :name,
+              :authors,
+              :books,
+              :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @check_out_count = Hash.new(0)
+  end
+
+  def add_author(author)
+    @authors << author
+    author.books.each do |book|
+      @books << book
+    end
+  end
+
+  def publication_time_frame_for(author)
```

Comment:
The logic here is solid, but what might it look like if an author was responsible for knowing their earliest and latest books?  Or maybe responsible for compiling a list of books from earliest to latest?